Thank you! I see that you also implemented a `Chimeric` type.

The tests pass for me locally on Python 3.12 but they do fail on 3.11:

```python
$ python -m unittest test_proforma.ProFormaTest.test_fragments
E
======================================================================
ERROR: test_fragments (test_proforma.ProFormaTest.test_fragments)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/levitsky/pyteomics/tests/test_proforma.py", line 166, in test_fragments
    for o, e in zip(i.fragments("b"), b_expected):
                    ^^^^^^^^^^^^^^^^
  File "/home/levitsky/pyteomics/pyteomics/proforma.py", line 4222, in fragments
    intervals = sorted(intervals, key=lambda x: x.start, reverse=reverse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object cannot be interpreted as an integer

----------------------------------------------------------------------
Ran 1 test in 0.002s

FAILED (errors=1)
```

--- a/pyteomics/proforma.py
+++ b/pyteomics/proforma.py
@@ -3251,6 +3251,9 @@ class Parser:
                 self.index,
                 self.state,
             )
+        for v in self.intervals:
+            assert v.start is not None
+            assert v.end is not None
         return self.positions, {
             "n_term": self.n_term,
             "c_term": self.c_term,

--- a/pyteomics/proforma.py
+++ b/pyteomics/proforma.py
@@ -16,7 +16,7 @@ import warnings
 
 from typing import Any, Callable, Dict, Iterable, Iterator, List, Optional, ClassVar, Sequence, Tuple, Type, Union, Generic, TypeVar, NamedTuple, overload, Literal
 from collections import Counter, deque, namedtuple
-from functools import partial
+from functools import partial, total_ordering
 from itertools import chain
 from array import array as _array
 from enum import Enum
@@ -1943,7 +1943,7 @@ class IntersectionEnum(Enum):
     start_overlap = 3
     end_overlap = 4
 
-
+@total_ordering
 class TaggedInterval(object):
     '''Define a fixed interval over the associated sequence which contains the localization
     of the associated tag or denotes a region of general sequence order ambiguity.
@@ -1979,6 +1979,37 @@ class TaggedInterval(object):
             [v.copy() for v in self.tags] if self.tags else [],
             self.ambiguous
         )
+
+    def __lt__(self, other: "TaggedInterval"):
+        if other is None:
+            return False
+        if self.start is None:
+            if other.start is None:
+                if self.end is None:
+                    if other.end is None:
+                        return False
+                    else:
+                        return True
+                elif other.end is None:
+                    return False
+                else:
+                    return self.end < other.end
+            else:
+                return True
+        else:
+            if other.start is None:
+                return False
+            elif self.start == other.start:
+                if self.end is None:
+                    if other.end is None:
+                        return False
+                    else:
+                        return True
+                elif other.end is None:
+                    return False
+                else:
+                    return self.end < other.end
+        return self.start < other.start
 
     def __eq__(self, other):
         if other is None:
@@ -3767,7 +3798,7 @@ def to_proforma(
         else:
             primary.append(str(aa) + "".join(["[{0!s}]".format(t) for t in tags]))
     if intervals:
-        for iv in sorted(intervals, key=lambda x: x.start):
+        for iv in sorted(intervals):
             if iv.ambiguous:
                 primary[iv.start] = "(?" + primary[iv.start]
             else:
@@ -4219,7 +4250,7 @@ class ProForma(object):
 
         intervals = self.intervals
         if intervals:
-            intervals = sorted(intervals, key=lambda x: x.start, reverse=reverse)
+            intervals = sorted(intervals, reverse=bool(reverse))
         intervals = deque(intervals)
 
         if include_labile:

--- a/pyteomics/proforma.py
+++ b/pyteomics/proforma.py
@@ -3251,9 +3251,6 @@ class Parser:
                 self.index,
                 self.state,
             )
-        for v in self.intervals:
-            assert v.start is not None
-            assert v.end is not None
         return self.positions, {
             "n_term": self.n_term,
             "c_term": self.c_term,

--- a/tests/test_proforma.py
+++ b/tests/test_proforma.py
@@ -169,6 +169,9 @@ class ProFormaTest(unittest.TestCase):
                 d += 204
             self.assertAlmostEqual(o, e + d, 3)
         self.assertEqual(d, 408)
+        for v in i.intervals:
+            assert v.start is not None
+            assert v.end is not None
 
     def test_slice(self):
         seq = ProForma.parse('[U:1]-MPEP-[UNIMOD:2]/2')

